File: MiniDOMBase/DMD/Anagrafica/Distributori/DistributoriFonteProvider.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
     
 
    Public Class DistributoriFonteProvider
        Implements IFonteProvider

        Public Function GetItemById(ByVal nome As String, id As Integer) As IFonte Implements IFonteProvider.GetItemById
            Return Anagrafica.Distributori.GetItemById(id)
        End Function

        Public Function GetItemByName(ByVal tipo As String, ByVal name As String) As IFonte Implements IFonteProvider.GetItemByName
            Return Anagrafica.Distributori.GetItemByName(name)
        End Function

        Public Function GetItemsAsArray(ByVal nome As String, Optional onlyValid As Boolean = True) As IFonte() Implements IFonteProvider.GetItemsAsArray
            Dim ret As New CCollection(Of IFonte)
            Dim cursor As New CDistributoriCursor
            cursor.OnlyValid = onlyValid
            cursor.Stato.Value = ObjectStatus.OBJECT_VALID
            cursor.Nome.SortOrder = SortEnum.SORT_ASC
            While Not cursor.EOF
                ret.Add(cursor.Item)
                cursor.MoveNext()
            End While
            cursor.Dispose()
            Return ret.ToArray
        End Function


        Public Function GetSupportedNames() As String() Implements IFonteProvider.GetSupportedNames
            Return New String() {"Distributori"}
        End Function
    End Class


End Class

File: MiniDOMBase/DMD/Anagrafica/Distributori/Providers/DistributoriFonteProvider.cs

using DMD.Databases;
using System.Collections.Generic;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Rappresenta una sorgente di fonti
        /// </summary>
        public class DistributoriFonteProvider 
            : IFonteProvider
        {

            /// <summary>
            /// Restituisce la fonte in base all'id
            /// </summary>
            /// <param name="nome"></param>
            /// <param name="id"></param>
            /// <returns></returns>
            public IFonte GetItemById(string nome, int id)
            {
                return Distributori.GetItemById(id);
            }

            /// <summary>
            /// Restituisce la fonte in base all'id
            /// </summary>
            /// <param name="tipo"></param>
            /// <param name="name"></param>
            /// <returns></returns>
            public IFonte GetItemByName(string tipo, string name)
            {
                return Distributori.GetItemByName(name);
            }

            /// <summary>
            /// Restituisce le fonti in base al tipo
            /// </summary>
            /// <param name="nome"></param>
            /// <param name="onlyValid"></param>
            /// <returns></returns>
            public IFonte[] GetItemsAsArray(string nome, bool onlyValid = true)
            {
                var ret = new List<IFonte>();
                using (var cursor = new CDistributoriCursor())
                {
                    cursor.OnlyValid = onlyValid;
                    cursor.Stato.Value = ObjectStatus.OBJECT_VALID;
                    cursor.Nome.SortOrder = SortEnum.SORT_ASC;
                    while (cursor.Read())
                    {
                        ret.Add(cursor.Item);
                    }
                }
                return ret.ToArray();
            }

            /// <summary>
            /// Restituisce i tipo di fonte supportati
            /// </summary>
            /// <returns></returns>
            public string[] GetSupportedNames()
            {
                return new string[] { "Distributori" };
            }
        }
    }
}